switchmap/server/db/table/metrics.py

```
@@ -0,0 +1,164 @@
+        try:
+            s = value.strip()
+            if s.endswith("Z"):
+                s = s[:-1] + "+00:00"
+            dt = datetime.datetime.fromisoformat(s)
+            if dt.tzinfo is None:
+                dt = dt.replace(tzinfo=datetime.timezone.utc)
+            return max(0, int(dt.timestamp()))
+        except ValueError as exc:
+            raise TypeError(
+                "Invalid last_polled ISO-8601 string"
+            ) from exc  # noqa: TRY003
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)